File: Week 1 C/Problem Sets/Problem Set 1/hello/hello.c
/**
 * @file hello.c
 * @brief CS50x Problem Set 1 — Hello: greets the user by name.
 *
 * @author Ayman Elmasry — AEL Digital Studio
 * @project AEL Sovereign — CS50x 2026-2027
 *
 * @details
 *   The canonical first program of the course. It demonstrates two of the
 *   most fundamental building blocks of C: reading input through the CS50
 *   library's `get_string` and writing formatted output through `printf`.
 *   No command-line arguments are involved; the program is entirely driven
 *   by an interactive prompt.
 *
 * Algorithm:
 *   1. Prompt the user with "What is your name? " and read a full line.
 *   2. Emit a single, personalized greeting line via one printf call.
 *
 * Complexity:
 *   Time  — O(1): one prompt and one printf, independent of name length.
 *   Space — O(1): only a pointer to the library-managed string is retained.
 */

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    /* Ask for the user's name and remember it for the greeting. */
    string name = get_string("What is your name? ");

    /* Print the required "hello, <name>" greeting line. */
    printf("hello, %s\n", name);

    return 0;
}

File: Week 6 Python/07_Problem_Sets/Problem Set 6/sentimental-hello/hello.py
#!/usr/bin/env python3
# ==============================================================================
# Project   : AEL Sovereign — CS50x 2026-2027
# Module    : week6_sentimental_hello
# File      : hello.py
# Author    : Ayman Elmasry — AEL Digital Studio
# ------------------------------------------------------------------------------
# Description:
#   Greets the user by name. Reads a single line from standard input and echoes
#   a personalized "hello, <name>" greeting — the canonical CS50 hello program
#   re-expressed in Python.
#
# Objectives:
#   - Demonstrate prompt-driven stdin interaction.
#   - Emphasise idiomatic f-string interpolation over legacy concatenation.
#   - Guard the entry point with the `__main__` idiom for reusability/import.
#
# Complexity:
#   Time : O(n) — walking the input string during the f-string read/format.
#   Space: O(n) — the name string held in memory for the duration of main().
# ==============================================================================


def main() -> None:
    """
    Prompt the user for their name and print a personalised greeting.
    The prompt deliberately ends with a newline, mirroring the exact
    check50-expected behaviour of the standard CS50 hello program.
    """
    name = input("What is your name?\n")
    print(f"hello, {name}")


if __name__ == "__main__":
    main()